### PR TITLE
ripped out FAQ

### PR DESCRIPTION
**Pull request by jbloom22**
I've culled the herd to a list of content to re-add to other places, hope to finish that later today so this can be merged.

### CODE REVIEW

File: docs/tutorial/Tutorial.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ In this window, enter two commands:
     >>> from hail import *
     >>> hc = HailContext()
 
-If there is no error, you're ready to start using Hail! Otherwise, make sure that the `export` variables are correctly set and appropriate versions of all dependencies are installed. If this step triggers an `Exception in thread "main" java.net.BindException`, see [here](https://hail.is/faq.html#how-do-i-fix-exception-in-thread-main-java.net.bindexception-cant-assign-requested-address-...) for a fix.
+If there is no error, you're ready to start using Hail! Otherwise, make sure that the `export` variables are correctly set and appropriate versions of all dependencies are installed. If this step triggers an `Exception in thread "main" java.net.BindException`, see FIXME
```
Comment (jigold) [inline docs/tutorial/Tutorial.md]: Fill in FIXME

File: python/hail/dataset.py [hunk 2]
```diff
@@ -648,12 +655,15 @@ def annotate_samples_table(self, input, sample_expr, root=None, code=None, confi
     def annotate_samples_vds(self, right, root=None, code=None):
         """Annotate samples with sample annotations from .vds file.
 
-        :param VariantDataset right: VariantDataset to annotate with.
+        :param right: VariantDataset to annotate with.
```
Comment (jigold) [inline python/hail/dataset.py]: Why is this changed?

File: python/hail/dataset.py [hunk 3]
```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
...
+        - contig lines
+        - lines added by external tools (bcftools, GATK, ...)
+
+        Hail exports the contents of `va.info` to the INFO field, so the workflow
+
+        >>> hc.import_vcf('data/example.vcf').export_vcf('data/example2.vcf')
+
+        will write out the same INFO field that was read in.
+
+        If samples or genotypes are filtered after importing a VCF, the value stored in `va.info.AC` value may no longer reflect the number of called alternate alleles in the filtered VDS. If the filtered VDS is then exported to VCF, downstream tools may produce erroneous results. The solution is to create new annotations in `va.info` or overwrite existing annotations. For example, in order to produce an accurate "AC" field, one can run py:meth:`variant_qc` and copy the `va.qc.AC` field to `va.info.AC`:
```
Comment (jigold) [inline python/hail/dataset.py]: code is double backticks in rst

File: python/hail/dataset.py [hunk 4]
```diff
@@ -1755,24 +1822,33 @@ def filter_variants_expr(self, condition, keep=True):
...
 
 
-        Remove variants on chromosome 1:
+        Remove all variants on chromosome 1:
 
         >>> vds_filtered = (hc.read('data/example.vds')
         >>>                   .filter_variants_expr('v.contig == "1"',
         >>>                                          keep=False))
 
+        Keep variants on chromosomes 1, 2, and 3:
```
Comment (jigold) [inline python/hail/dataset.py]: This is duplicated.

File: python/hail/dataset.py [hunk 5]
```diff
@@ -382,15 +382,22 @@ def annotate_samples_expr(self, condition):
...
+        >>>    .annotate_variants_table('data/consequence.tsv', 'Variant', code='va.consequence = table.Consequence')
+        >>>    .annotate_variants_expr('va.isSingleton = gs.map(g => g.nNonRefAlleles).sum() == 1')
+        >>>    .annotate_samples_expr('sa.LOF_genes = gs.filter(g => va.isSingleton && g.isHet && va.consequence == "LOF").map(g => va.gene).collect()'))
+
+        To create an annotation for only a subset of samples based on an existing annotation:
+
+        >>> (hc.read('data/example.vds')
+        >>>    .annotate_samples_expr('if (sa.pheno.cohortName == "cohort1") sa.pheno.bloodPressure else NA: Double')
+
+        For optimal performance, make sure the Type defined after NA is the same as the type of ``sa.pheno.bloodPressure``.
```
Comment (jigold) [inline python/hail/dataset.py]: maybe use .. note: (look up exact syntax)

File: python/hail/dataset.py [hunk 6]
```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
...
+
+        Export to VCF as an uncompressed file:
+
+        >>> hc.read('data/example.vds').export_vcf('data/example.vcf')
+
+        Export to VCF as block-compressed files with linear-scaled dosages:
+
+        >>> hc.read('data/example.vds').export_vcf('data.example.vcf.bgz', export_pp=True, parallel=True)
+
+        Export a sites-only VCF with addition header lines from `myheader.txt`:
```
Comment (jigold) [inline python/hail/dataset.py]: additional

File: python/hail/dataset.py [hunk 7]
```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
...
+
+        >>> (hc.read('data/example.vds')
+        >>>    .filter_samples_all()
+        >>>    .export_vcf('data/sites-only.vcf.bgz', append_to_header='data/myheader.txt')
+
+        **Notes**
+        py:meth:`exportvcf` exports the VDS the VCF format as described in the [VCF 4.2 spec](https://samtools.github.io/hts-specs/VCFv4.2.pdf).
```
Comment (jigold) [inline python/hail/dataset.py]: formatting here!

Comment (jigold) [inline python/hail/dataset.py]: change weblink to RST

File: python/hail/dataset.py [hunk 8]
```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
...
+
+        - Use the ``.vcf.bgz`` extension in the output file name for `blocked GZIP <http://www.htslib.org/doc/tabix.html>`_ compression. We strongly recommended this for exporting large VCFs.
+        - Set ``parallel=True`` to return a set of VCF files (one per partition) rather than serially concatenating these files.
```
Comment (jigold) [inline python/hail/dataset.py]: Why aren't these in the argument descriptions?

File: python/hail/dataset.py [hunk 9]
```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
...
+
+        If samples or genotypes are filtered after importing a VCF, the value stored in `va.info.AC` value may no longer reflect the number of called alternate alleles in the filtered VDS. If the filtered VDS is then exported to VCF, downstream tools may produce erroneous results. The solution is to create new annotations in `va.info` or overwrite existing annotations. For example, in order to produce an accurate "AC" field, one can run py:meth:`variant_qc` and copy the `va.qc.AC` field to `va.info.AC`:
+
+        >>> (hc.import_vcf('data/example.vcf')
+        >>>    .filter_genotypes('g.gq >= 20')
+        >>>    .variant_qc()
+        >>>    .annotate_variants_expr('va.info.AC = va.qc.AC')
+        >>>    .export_vcf('data/example.vcf.bgz'))
+
+        `py:meth:`export_vcf` does not export any annotations besides those in `va.info`.
```
Comment (jigold) [inline python/hail/dataset.py]: fix link

File: python/hail/dataset.py [hunk 10]
```diff
@@ -1755,24 +1822,33 @@ def filter_variants_expr(self, condition, keep=True):
...
         >>> vds_filtered = (hc.read('data/example.vds')
         >>>                   .filter_variants_expr('v.contig == "1"',
         >>>                                          keep=False))
 
+        Keep variants on chromosomes 1, 2, and 3:
+
+        >>> vds_filtered = (hc.read('data/example.vds')
+        >>>                   .filter_variants_expr('set("1","2","3").contains(v.contig)')
+
+        Keep variants on chromosomes 1, 2, and 3:
```
Comment (jigold) [inline python/hail/dataset.py]: I'd keep only 1 of these examples.

File: python/hail/dataset.py [hunk 11]
```diff
@@ -1755,24 +1822,33 @@ def filter_variants_expr(self, condition, keep=True):
...
 
         - ``v`` (*Variant*): :ref:`variant`
         - ``va``: variant annotations
         - ``global``: global annotations
         - ``gs`` (*Aggregable[Genotype]*): aggregable of :ref:`genotype` for variant ``v``
 
-        For more information, see the documentation on `data representation, annotations <overview.html#>`_, and
-        the `expression language <../expr_lang.html>`_.
+        .. caution::
+        ``va.contig`` is of type String.
```
Comment (jigold) [inline python/hail/dataset.py]: Need to tab this line

Comment (jigold) [inline python/hail/dataset.py]: Also this makes no sense to me. Did you mean `v.contig`? Not sure why this is here.

File: python/hail/dataset.py [hunk 12]
```diff
@@ -2464,22 +2540,22 @@ def mendel_errors(self, output, fam):
         """Find Mendel errors; count per variant, individual and nuclear
         family.
 
-        **Implementation Details**
+        **Example**
```
Comment (jigold) [inline python/hail/dataset.py]: Examples

File: python/hail/dataset.py [hunk 13]
```diff
@@ -2464,22 +2540,22 @@ def mendel_errors(self, output, fam):
...
 
-        This method finds all violations of Mendelian inheritance in each (dad,
-        mom, kid) trio of samples.
+        Find all violations of Mendelian inheritance in each (dad,
+        mom, kid) trio in `trios.fam` and save results to files with root `mydata`:
 
-        The following expression,
+        >>> hc.read(sample.vds)..mendel_errors('mydata', 'trios.fam')
```
Comment (jigold) [inline python/hail/dataset.py]: this has extra period

File: python/hail/dataset.py [hunk 14]
```diff
@@ -2464,22 +2540,22 @@ def mendel_errors(self, output, fam):
         """Find Mendel errors; count per variant, individual and nuclear
         family.
 
-        **Implementation Details**
+        **Example**
 
-        This method finds all violations of Mendelian inheritance in each (dad,
-        mom, kid) trio of samples.
+        Find all violations of Mendelian inheritance in each (dad,
+        mom, kid) trio in `trios.fam` and save results to files with root `mydata`:
```
Comment (jigold) [inline python/hail/dataset.py]: Use asterisks for italics rather than back ticks.

File: python/hail/docs/hail_objects.rst [hunk 15]
```diff
@@ -22,8 +22,8 @@ Variant
 - **v.nAltAlleles** (*Int*) -- Number of alternate alleles, equal to ``nAlleles - 1``
 - **v.nGenotypes** (*Int*) -- Number of genotypes
 - **v.altAlleles** (*Array[AltAllele]*) -- The :ref:`alternate alleles <altallele>`
-- **v.inXPar** (*Boolean*) -- True if chromosome is X and start is in pseudo-autosomal region of X
-- **v.inYPar** (*Boolean*) -- True if chromosome is Y and start is in pseudo-autosomal region of Y. *NB: most callers assign variants in PAR to X*
+- **v.inXPar** (*Boolean*) -- True if chromosome is X and start is in :ref:`pseudoautosomal region <pseduoautosomal_region>` of X
```
Comment (jigold) [inline python/hail/docs/hail_objects.rst]: These links aren't showing up on the CI.

File: python/hail/dataset.py [hunk 16]
```diff
@@ -1755,24 +1822,33 @@ def filter_variants_expr(self, condition, keep=True):
...
+        ``condition`` is in the variant context so the following symbols are in scope:
 
         - ``v`` (*Variant*): :ref:`variant`
         - ``va``: variant annotations
         - ``global``: global annotations
         - ``gs`` (*Aggregable[Genotype]*): aggregable of :ref:`genotype` for variant ``v``
 
-        For more information, see the documentation on `data representation, annotations <overview.html#>`_, and
-        the `expression language <../expr_lang.html>`_.
+        .. caution::
```
Comment (jigold) [inline python/hail/dataset.py]: Also, why did you delete the links for more information?

File: python/hail/dataset.py [hunk 17]
```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
...
+        - Set ``parallel=True`` to return a set of VCF files (one per partition) rather than serially concatenating these files.
+        - Set ``append_to_header=file.txt`` to append *file.txt* to the output VCF header.
+        - Set ``export_pp=True`` to export linear-scaled probabilities (Hail's pp field on genotype) as the VCF PP FORMAT field.
+
+        Importing a VCF to a Hail VDS and then exporting to a VCF will not result in exactly the same VCF.  The differences, however, are limited to the header and INFO field.  The output VCF header will contain:
+
+        - FORMAT lines, with the format "GT:AD:DP:GQ:PL", or "GT:AD:DP:GQ:PP" if ``export_pp=True``
+        - FILTER lines, if present in the imported VCF
+        - INFO lines (see exported annotations below)
+
```
Comment (jigold) [inline python/hail/dataset.py]: It would be nice if you could format this section better. Super difficult to read.

File: docs/tutorial/Tutorial.md (hunk 1 above)
Comment (jbloom22) [inline docs/tutorial/Tutorial.md]: Removed the sentence. Dan is going to make a discuss post and then add a link to it.

File: python/hail/dataset.py (hunk 12 above)
Comment (jbloom22) [inline python/hail/dataset.py]: Okay, I've changed all such sections to **Examples** no matter if it's one or more example.

**Comment by jbloom22** [comment]
Also made the line spacing of parameters and before/after """ consistent throughout the python docs.

**Comment by jbloom22** [comment]
Back to you!

File: python/hail/dataset.py [hunk 18]
```diff
@@ -1306,7 +1342,7 @@ def export_variants(self, output, condition, types=None):
 
         .. note::
 
-            Either all fields must be named, or no field must be named.
+            If any field is named, all fields must be names.
```
Comment (jigold) [inline python/hail/dataset.py]: typo: names -> named

File: python/hail/dataset.py [hunk 19]
```diff
@@ -1433,53 +1471,51 @@ def export_vcf(self, output, append_to_header=None, export_pp=False, parallel=Fa
...
-        Export a sites-only VCF with addition header lines from `myheader.txt`:
+        Export a sites-only VCF with additional header lines from *myheader.txt*:
 
         >>> (hc.read('data/example.vds')
         >>>    .filter_samples_all()
         >>>    .export_vcf('data/sites-only.vcf.bgz', append_to_header='data/myheader.txt')
 
         **Notes**
-        py:meth:`exportvcf` exports the VDS the VCF format as described in the [VCF 4.2 spec](https://samtools.github.io/hts-specs/VCFv4.2.pdf).
+        py:meth:`exportvcf` writes the VDS to disk in VCF format as described in the `VCF 4.2 spec <https://samtools.github.io/hts-specs/VCFv4.2.pdf>`_.
```
Comment (jigold) [inline python/hail/dataset.py]: This link is still messed up. Maybe forgetting a tilde or a period?

File: python/hail/dataset.py [hunk 20]
```diff
@@ -1433,53 +1471,51 @@ def export_vcf(self, output, append_to_header=None, export_pp=False, parallel=Fa
 
         >>> hc.read('data/example.vds').export_vcf('data.example.vcf.bgz', export_pp=True, parallel=True)
```
Comment (jigold) [inline python/hail/dataset.py]: I'd remove the last two examples (too specialized) and make the example above for a block compressed vcf.

File: python/hail/dataset.py [hunk 21]
```diff
@@ -1433,53 +1471,51 @@ def export_vcf(self, output, append_to_header=None, export_pp=False, parallel=Fa
...
-        - FILTER lines, if present in the imported VCF
-        - INFO lines (see exported annotations below)
+        Following import-export workflow
 
-        The output VCF header will not contain these lines unless they are explicitly copied to a file and inserted using the ``append_to_header`` option:
+        >>> hc.import_vcf('data/example.vcf').export_vcf('data/example_out.vcf')
 
-        - contig lines
-        - lines added by external tools (bcftools, GATK, ...)
+        the *example_out.vcf* header will contain the FORMAT, FILTER, and INFO lines present in *example.vcf*.  However, it will *not* contain CONTIG lines or lines added by external tools (such as bcftools and GATK) unless they are explicitly inserted using the ``append_to_header`` option.
```
Comment (jigold) [inline python/hail/dataset.py]: I still think this section is incredibly confusing. Maybe use bullet points and reduce the number of examples.

File: python/hail/dataset.py [hunk 22]
```diff
@@ -1433,53 +1471,51 @@ def export_vcf(self, output, append_to_header=None, export_pp=False, parallel=Fa
...
+        The solution is to create new annotations in ``va.info`` or overwrite existing annotations. For example, in order to produce an accurate ``AC`` field, one can run py:meth:`variant_qc` and copy the ``va.qc.AC`` field to ``va.info.AC``:
 
         >>> (hc.import_vcf('data/example.vcf')
         >>>    .filter_genotypes('g.gq >= 20')
         >>>    .variant_qc()
         >>>    .annotate_variants_expr('va.info.AC = va.qc.AC')
         >>>    .export_vcf('data/example.vcf.bgz'))
 
-        `py:meth:`export_vcf` does not export any annotations besides those in `va.info`.
+        py:meth:`export_vcf` does not export any annotations besides those in ``va.info``.
```
Comment (jigold) [inline python/hail/dataset.py]: Link still broken

File: python/hail/dataset.py (hunk 19 above)
Comment (jbloom22) [inline python/hail/dataset.py]: I just checked and both links look and work fine in the built docs.  Not sure what you're seeing.

Comment (jbloom22) [inline python/hail/dataset.py]: Oh, you didn't mean the weblinks.  okay

File: python/hail/dataset.py (hunk 21 above)
Comment (jbloom22) [inline python/hail/dataset.py]: I tried to make it better.

**Comment by jbloom22** [comment]
Back to you!

**Comment by jbloom22** [comment]
Killed errant code that snuck into this branch, tests passing again.
